hyrisecockpit/server.py

```
@@ -42,16 +40,13 @@ def start(self) -> None:
             self._socket.send_json(response)
 
     def _handle_request(self, request: Request) -> Response:
-        try:
-            validate(instance=request, schema=request_schema)
-            func, schema = self._calls[request["header"]["message"]]
-            if schema:
-                validate(request["body"], schema=schema)
-            return func(request["body"])
-        except ValidationError:
-            return get_response(400)
-        except KeyError:
+
+        # TODO remove validation schema and call not found. We handle all this stuff in flask.
```

A server should still validate that requests are at least readable and drop them if they are not. This makes the server more robust and is in my opinion a super low hanging fruit.

---

Because this server only communicates with the flask application and this application is well testet I don't see the need to validate the message. For development porpoise I think it is even better to let the application crash if a message is not valid to find the bug quickly. For me a seperate validation makes sense if we don't know how is communicating with it.

---

> Because this server only communicates with the flask application and this application is well testet I don't see the need to validate the message.

The communication between flask and the dbm/wlg is not well tested.

> For development porpoise I think it is even better to let the application crash if a message is not valid to find the bug quickly.

Sure, we could add a `validate: bool` flag that could be easily be switched off when developing.

Besides, I don't see how this change is crucial to adding TPC-C?